src/NuGet.Clients/NuGet.VisualStudio.Common/ServiceLocator.cs

```diff
@@ -27,49 +25,6 @@ public static void InitializePackageServiceProvider(IAsyncServiceProvider provid
 
         private static IAsyncServiceProvider PackageServiceProvider;
 
-        /// <inheritdoc cref="GetInstanceAsync{TService}"/>
-        public static TService GetInstance<TService>() where TService : class
-        {
-            return NuGetUIThreadHelper.JoinableTaskFactory.Run(GetInstanceAsync<TService>);
-        }
-
-        /// <summary>
-        /// Fetches a service that may be registered with the DTE, MEF or the VS service provider.
-        /// This method may switch to the UI thread.
-        /// </summary>
-        /// <typeparam name="TService"></typeparam>
-        /// <returns>The instance of the service request, <see langword="null"/> otherwise. </returns>
-        /// <remarks>
-        /// Prefer <see cref="GetComponentModelServiceAsync{TService}{TService}"/> over this requesting a MEF service.
-        /// A general rule is that only non-NuGet VS services should be retrieved this method.
-        /// </remarks>
-        public static async Task<TService> GetInstanceAsync<TService>() where TService : class
-        {
-            // Try to find the service as a component model, then try dte then lastly try global service
-            // Per bug #2072, avoid calling GetGlobalService() from within the Initialize() method of NuGetPackage class.
-            // Doing so is illegal and may make VS to stop responding. As a result of that, we defer calling GetGlobalService to the last option.
-
-            var serviceFromDTE = await GetDTEServiceAsync<TService>();
-            if (serviceFromDTE != null)
-            {
-                return serviceFromDTE;
-            }
-
-            var serviceFromComponentModel = await GetComponentModelServiceAsync<TService>();
-            if (serviceFromComponentModel != null)
-            {
-                return serviceFromComponentModel;
-            }
-
-            var globalService = await GetGlobalServiceAsync<TService, TService>();
-            return globalService;
-        }
-
-        public static TInterface GetGlobalService<TService, TInterface>() where TInterface : class
-        {
-            return NuGetUIThreadHelper.JoinableTaskFactory.Run(GetGlobalServiceAsync<TService, TInterface>);
-        }
-
         public static async Task<TInterface> GetGlobalServiceAsync<TService, TInterface>() where TInterface : class
```

Comment:
If I'm not mistaken, then this one resolves to MEF right?
Should we add small comment to this one to make clear for other developers?

Reply:
No, It's not MEF. This hits the service provider.
 GetComponentModelServiceAsync is MEF.
 I added some docs.